lib/notificationScheduler.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+const co = require('co')
+const defer = require('co-defer')
+const MAX_32INT = 2147483647
+
+/**
+ * @param {Object} options
+ * @param {Class} options.Notification
+ * @param {Object} options.knex
+ * @param {Logger} options.log
+ * @param {function(notification) -> Promise} options.processNotification
+ */
+function NotificationScheduler (options) {
+  this.Notification = options.Notification
+  this.knex = options.knex
+  this.log = options.log
+  this.processNotification = options.processNotification
+  this._timeout = null
+}
+
+NotificationScheduler.prototype.isEnabled = function () { return !!this._timeout }
+
+NotificationScheduler.prototype.start = function () {
+  if (this._timeout) return
+  this._timeout = defer.setTimeout(this.processQueue.bind(this), 0)
+}
+
+NotificationScheduler.prototype.stop = function () {
+  if (!this._timeout) return
+  clearTimeout(this._timeout)
+  this._timeout = null
+}
+
+NotificationScheduler.prototype.scheduleProcessing = co.wrap(function * () {
+  if (!this._timeout) return
+  clearTimeout(this._timeout)
+  const timeToEarliestNotification = yield this._getTimeToEarliestNotification()
+  if (!timeToEarliestNotification) return
```

Comment:
Since the NotificationScheduler isn't actually watching new entries added, the users of this functionality need to be sure that they call `start` every time they add something new that will need notifications